### PR TITLE
v1.6 backports 2020-03-04

### PR DESCRIPTION
**Pull request by vadorovsky**
* #10388 -- test: Avoid using global map for Cilium configuration (@brb)
 * #10262 -- [CI] Replace jenkinsfiles with symlinks (@nebril)

*Not backported as they're handled by #10638:*

 * #10218 -- cilium: encryption + tunnel improvements and fixes (@jrfastab)
 * #10268 -- cilium: encryption, segfaults if existing non-Cilium xfrm policy without mark set exists (@jrfastab)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 10388 10262; do contrib/backporting/set-labels.py $pr done 1.6; done
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10460)

### CODE REVIEW

**Comment by mrostecki** [comment]
never-tell-me-the-odds

**Comment by mrostecki** [comment]
never-tell-me-the-odds

**Comment by mrostecki** [comment]
never-tell-me-the-odds

**Comment by aanm** [comment]
test-me-please

File: test/k8sT/DatapathConfiguration.go [hunk 1]
```diff
@@ -154,25 +154,28 @@ var _ = Describe("K8sDatapathConfig", func() {
...
 		It("Check direct connectivity with per endpoint routes", func() {
 			SkipIfFlannel()
 
-			deployCilium(append(directRoutingOptions,
+			deployCilium([]string{
+				"--set global.tunnel=disabled",
+				"--set global.autoDirectNodeRoutes=true",
 				"--set global.endpointRoutes.enabled=true",
-			))
+				"--set global.ipv6.enabled=false",
```
Comment (brb) [inline test/k8sT/DatapathConfiguration.go]: It seems that this was unintentionally added (IPv6 was not previously disabled in this test in v1.6)?

**Review by brb — CHANGES_REQUESTED** [review]

File: test/k8sT/DatapathConfiguration.go (hunk 1 above)
Comment (vadorovsky) [reply test/k8sT/DatapathConfiguration.go]: Yes, true. I though I should stick to parameters you have in #10388, but I can remove it.

**Comment by mrostecki** [comment]
test-me-please

**Review by brb — APPROVED** [review]

**Comment by mrostecki** [comment]
test-me-please

**Comment by mrostecki** [comment]
never-tell-me-the-odds

**Comment by joestringer** [comment]
The test `K8sDatapathConfig Transparent encryption DirectRouting Check connectivity with transparent encryption and direct routing` seems to be reliably failing:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/17774/

Given that this is related to a PR that's being backported as part of this PR, I think it's likely to be related and we should dig into this further before merging this PR.

**Comment by joestringer** [comment]
I presume that @mrostecki @jrfastab will co-ordinate on moving forward on this PR (whether by dropping the backport or figuring out how to address it on v1.6).

**Comment by joestringer** [comment]
I just dropped the encryption backport commits in light of https://github.com/cilium/cilium/pull/10638 .

**Comment by joestringer** [comment]
never-tell-me-the-odds

**Comment by joestringer** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]

**Pull request merged**